File: AzW.Web.Portal/src/models/Function.js
import ResourceType from './ResourceType'
export default class Function
{
    constructor (){
        this.GraphModel = {
            Id: '',
            ResourceType: ResourceType.Function(),
            DisplayName: ''
        };

        this.ProvisionContext = {
            ResourceType: ResourceType.Function(),
            Name: 'func-web',
            Location: 'westus',
            ResourceGroupName: '',
            IsConsumptionPlan: true,
            IsLinux: false,
            PricingTier: ''
        };
        this.CalculatorContext = {
            IsFree: false,
            Tier: ''
        }
    }
}

File: Portal.Services/AzW.Model/Provision/Function.cs
namespace AzW.Model
 {
     public class Function
     {
          public string Name { get; set; }
          public string Location { get; set; }
          public string ResourceGroupName { get; set; }
          public bool IsConsumptionPlan { get; set; }
          public bool IsLinux { get; set; }
          public string PricingTier { get; set; }
     }
 }